src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -94,14 +95,9 @@ class CornerstoneViewport extends Component {
   constructor(props) {
     super(props);
 
-    const imageIdIndex = props.imageIdIndex;
-    const imageId = props.imageIds[imageIdIndex];
-
     this.state = {
       // Used for metadata lookup (imagePlane, orientation markers)
       // We can probs grab this once and hold on to? (updated on newImage)
-      imageId,
-      imageIdIndex, // Maybe
```

Comment:
The issue w/ removing this from `state` is that if `onNewImage` isn't used to update the `imageIdIndex` prop, then the overlay's `image` will be stuck at 1. We should be able to use this internally to display that value, and use the `prop` to "jump". If the `prop`'s updated value matches `state`, we can probably skip jumping.

Could also split props to make the `controlled` vs `uncontrolled` behavior more obvious:
- `initialImageIdIndex` - only used to set initial on mount
- `imageIdIndex` - if set, is controlled